koanf_test.go

```diff
@@ -436,12 +436,12 @@ func TestLoadMerge(t *testing.T) {
 
 	// Load env provider and override value.
 	os.Setenv("PREFIX_PARENT1.CHILD1.TYPE", "env")
-	err := k.Load(env.Provider("PREFIX_", ".", func(s string) string {
-		return strings.Replace(strings.ToLower(s), "prefix_", "", -1)
+	err := k.Load(env.ProviderWithValue("PREFIX_", ".", func(k string, v string) (string, interface{}) {
```

Comment:
The `env.Provider` test has been overwritten with `env.ProviderWithValue`. There should be tests for both functions. Could you re-instated the deleted test?